src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java

```diff
@@ -233,6 +233,16 @@ private void checkEnumConstant(DetailAST ast) {
      * @return true if ast is semicolon, false otherwise
      */
     private static boolean isSemicolon(DetailAST ast) {
-        return ast.getType() == TokenTypes.SEMI;
+        return ast != null && ast.getType() == TokenTypes.SEMI;
+    }
+
+    /**
+     * Wrapper method for isSemicolon, allowing testing from outside the package.
+     *
+     * @param ast the DetailAST node which needs to be checked
+     * @return true if the given ast is semicolon, false otherwise
+     */
+    public static boolean checkProcess(DetailAST ast) {
```

Comment:
Please use only Input file based testing. As all other tests are doing.

Reply:
I have been struggling with the folder structure. Right now, there are two input files in the `UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest` i.e. `InputUnnecessarySemicolonAfterTypeMemberDeclaration.java` located in `com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/InputUnnecessarySemicolonAfterTypeMemberDeclaration.java`

similarly, `InputUnnecessarySemicolonAfterTypeMemberDeclarationRecords.java` located in `src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/InputUnnecessarySemicolonAfterTypeMemberDeclarationRecords.java`

So, when i put my new input file inside any of the sub-directory of unnecessarysemicolonaftertypememberdeclaration it gives me error. I have two questions. Kindly bear me as i am contributing to this repo for the first time. 

Should I write the unit test in UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest or create a different class for it? If i am writing it in UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest, where do i place the unit test as there is very strong check on file name and location. Would something like this work?
```
unnecessarysemicolonaftertypememberdeclaration
 > semicolontests
   > InputIsSemicolonWithNullAst.java
 > InputUnnecessarySemicolonAfterTypeMemberDeclaration.java
```

Reply:
@romani I have pushed a new patchset. Kindly review.

Reply:
@relentless-pursuit , no worries, I know you are New. We and Ci will guide you through a process. Just  keep calm, and follow CI and us and we will succeed.

Reply:
Surely. Thanks for the assurance. And, I am currently going through the CI logs and making changes. I will push a new patchset once i have resolved most CI logs issue to the best of my capabilities. And, ask for further guidance if stuck. Thanks again.

Reply:
@Romani, finally after so many attempts from saturaday, I could see a green. Can you review and give me your feedback to proceed further?

Reply:
I tried understanding the failing issue mentioned in 'Check no closed issue references / check_issues'. But, i couldn't decipher. Any input on the same will be helpful.

Reply:
You need to remove module and comment at https://github.com/relentless-pursuit/checkstyle/blob/3b6d5b812082ebd5a81ecdfe5de1bbd0c50ae47c/config/projects-to-test/openjdk20-excluded.files#L40

Reply:
> You need to remove module and comment at https://github.com/relentless-pursuit/checkstyle/blob/3b6d5b812082ebd5a81ecdfe5de1bbd0c50ae47c/config/projects-to-test/openjdk20-excluded.files#L40

I assume the comment must be added by a script. As a beginnner who wants to keep contributing to this repo, how can we avoid it in the first place?

Reply:
Do not be afraid to use `git blame` / GitHub history to understand context and details about existing code. This module and corresponding comment were added in the scope of https://github.com/checkstyle/checkstyle/issues/13402

Reply:
@romani it seems to be passing all CIs. Can you review it once again?